services/searchengine/bleveengine/search.go

```diff
@@ -156,10 +156,23 @@ func (b *BleveEngine) SearchPosts(channels *model.ChannelList, searchParams []*m
 			}
 		} else {
 			if len(params.Terms) > 0 {
-				messageQ := bleve.NewMatchQuery(params.Terms)
-				messageQ.SetField("Message")
-				messageQ.SetOperator(termOperator)
-				termQueries = append(termQueries, messageQ)
+				terms := []string{}
+				for _, term := range strings.Split(params.Terms, " ") {
```

Comment:
Why do aggregate the queries in this for loop instead of doing here the wildcard query addition and after that the regular terms one?

Reply:
`params.Terms` can contain a value such as `sear* post` (in the test example), so first to distinguish if we have to run a wildcard query, we need to isolate each term and check if the wildcard is the suffix, and then, if we find a term with a wildcard, we have to exclude it from the normal `MatchQuery` in case the `termOperator` is and, because if it's treated as a literal, it will never be found